pkg/ccl/sqlproxyccl/server.go

```diff
@@ -186,6 +190,35 @@ func (s *Server) Serve(ctx context.Context, ln net.Listener) error {
 	}
 }
 
+// AwaitNoConnections returns a channel that is closed once the server has no open connections.
+// This is meant to be used after the server has stopped accepting new connections and we are
+// waiting to shutdown the server without inturrupting existing connections
+//
+// If the context is cancelled the channel will never close because we have to end the async task
+// to allow the stopper to completely finish
+func (s *Server) AwaitNoConnections(ctx context.Context) <-chan struct{} {
+	c := make(chan struct{})
+
+	_ = s.Stopper.RunAsyncTask(ctx, "await-no-connections", func(context.Context) {
```

Comment:
💯 This is a simpler implementation than I expected.

Reply:
🙌🏽